dungeon/src/task/quizquestion/Quiz.java

```diff
@@ -0,0 +1,152 @@
+     * @param questionContentType What does the question contain? Just text, just a path to an
+     *     image, or both text and a path to an image?
+     * @param questionText The question itself (can contain a path to images).
+     */
+    public Quiz(final QuizType type, QuizContentType questionContentType, String questionText) {
+        super();
+        this.type = type;
+        taskText(questionText);
+        question = new QuizContent(this, questionContentType, questionText);
+    }
+
+    /**
+     * Get the type of the question.
+     *
+     * @return The type of the question.
+     */
+    public QuizType type() {
+        return type;
+    }
+
+    /**
+     * Create a {@link QuizContent} answer instance with the given configuration as a possible
+     * answer for this question.
+     *
+     * @param type What does the answer contain? Just text, just a path to an image, or both text
+     *     and a path to an image?
+     * @param content The answer itself (can contain a path to images).
+     */
+    public void addAnswer(QuizContentType type, String content) {
+        addContent(new QuizContent(this, type, content));
+    }
+
+    /**
+     * Get the question instance.
+     *
+     * @return The question instance.
+     */
+    public QuizContent question() {
+        return question;
+    }
+
+    /**
+     * The QuizQuestionContentType enum represents the different types of content that can be
+     * associated with a quiz question. The available types are TEXT, IMAGE, and TEXT_AND_IMAGE.
+     * TEXT represents a question or answer choice as text. IMAGE represents a question or answer
+     * choice as an image. TEXT_AND_IMAGE represents a question or answer choice as both text and an
+     * image.
+     */
+    public enum QuizContentType {
+        TEXT,
+        IMAGE,
+        TEXT_AND_IMAGE
+    }
+
+    /**
+     * The QuizQuestionType enum represents the different types of quiz questions that can be
+     * created. The available types are SINGLE_CHOICE, MULTIPLE_CHOICE, and FREETEXT. SINGLE_CHOICE
+     * represents a question with multiple answer choices, where the user is required to select one
+     * answer. MULTIPLE_CHOICE represents a question with multiple answer choices, where the user is
+     * allowed to select multiple answers. FREETEXT represents a question where the user is required
+     * to input their answer as text.
+     */
+    public enum QuizType {
+        SINGLE_CHOICE,
+        MULTIPLE_CHOICE,
+        FREETEXT
+    }
+
+    /**
+     * Content for a {@link Quiz}-
+     *
+     * <p>Is used as answer and question for a {@link Quiz}.
+     *
+     * <p>Stores a String with the question/answer text and {@link QuizContentType} which defines if
+     * the String is just Text or contains a path to an image or both. The type is used by the
+     * {@link QuizUI} to configure the ui if the question is asked via the ui.
+     */
+    public static class QuizContent extends TaskContent {
```

Comment:
gibt es einen grund, dass wir hier keine normale klasse nutzen?

Reply:
Das `QuizContent` eine inner class von `Quiz` ist, ist eigentlich nur packaging convenience. Ich brauche keinen Zugriff auf die Objekt-Methoden von `Quiz`.
Steht auch so in der [Oracle doc](https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html#:~:text=Note%3A%C2%A0A,also%20demonstrates%20this.)

Was tatsächlich ein Argument gegen das static wäre, ist das non-static inner classes immer mit einer Instanz der äußeren klasse verbunden ist. So wirklich helfen tut mir das nicht, da ich das Quiz-Objekt eh in der Basisklasse `TaskContent` speicher, wäre also eher ein semantisches ding.

Reply:
ich sehe den gedanken hinter dem konstrukt, aber irgendwie wirft das in der verwendung einige fragen auf.